cleverhans/utils_th.py

```diff
@@ -62,6 +62,39 @@ def adadelta(loss_or_grads, params, learning_rate=1.0, rho=0.95, epsilon=1e-6):
 
     return updates
 
+def adam(loss_or_grads, params, learning_rate=0.001, beta1=0.9,
```

Comment:
the reason is to keep the code without external dependencies, as discussed [here](https://groups.google.com/d/msg/cleverhans-dev/BfYcQghdkwY/UWy9DHYuCAAJ). I think we need to explicitly write in the comments that the code is copied from lasagne. Importing lasagne only for the optimizers... i think it's an overkill, especially because [Lasagne docs suggest](http://lasagne.readthedocs.io/en/latest/user/installation.html#stable-lasagne-release) using non-pip version of Theano, which can become confusing.